### PR TITLE
[FLINK-18317][legal] Fix various licensing issues

### PR DESCRIPTION
**Pull request by zentol**
These can all be verified by checking the shade-plugin output.

### CODE REVIEW

File: flink-connectors/flink-sql-connector-hive-1.2.2/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -6,8 +6,8 @@ The Apache Software Foundation (http://www.apache.org/).
 
 This project bundles the following dependencies under the Apache Software License 2.0. (http://www.apache.org/licenses/LICENSE-2.0.txt)
 
+- io.airlift:aircompressor:0.8
 - org.apache.hive:hive-exec:1.2.2
 - org.apache.hive:hive-metastore:1.2.2
-- org.apache.thrift:libfb303:0.9.2
 - org.apache.orc:orc-core:1.4.3
-- io.airlift:aircompressor:0.8
+- org.apache.thrift:libfb303:0.9.2
```
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-1.2.2/src/main/resources/META-INF/NOTICE:13]: Note: There are more problems with this file: https://github.com/apache/flink/pull/12682

**Comment by zentol** [comment]
@rmetzger Please don't trust me on this; it is really easy to overlook something.

File: flink-connectors/flink-sql-connector-hive-1.2.2/src/main/resources/META-INF/NOTICE (hunk 1 above)
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-1.2.2/src/main/resources/META-INF/NOTICE:13]: ✅ checked.

File: flink-connectors/flink-sql-connector-hive-2.2.0/src/main/resources/META-INF/NOTICE [hunk 2]
```diff
@@ -6,6 +6,6 @@ The Apache Software Foundation (http://www.apache.org/).
 
 This project bundles the following dependencies under the Apache Software License 2.0. (http://www.apache.org/licenses/LICENSE-2.0.txt)
 
+- io.airlift:aircompressor:0.8
 - org.apache.hive:hive-exec:2.2.0
 - org.apache.orc:orc-core:1.4.3
-- io.airlift:aircompressor:0.8
```
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-2.2.0/src/main/resources/META-INF/NOTICE:11]: ✅

File: flink-dist/src/main/resources/META-INF/NOTICE [hunk 3]
```diff
@@ -12,7 +12,7 @@ This project bundles the following dependencies under the Apache Software Licens
 - com.twitter:chill_2.11:0.7.6
 - com.typesafe:config:1.3.0
 - com.typesafe:ssl-config-core_2.11:0.3.7
-- com.typesafe.akka:akka-actor_2.11:2.5.1
+- com.typesafe.akka:akka-actor_2.11:2.5.21
```
Comment (rmetzger) [inline flink-dist/src/main/resources/META-INF/NOTICE:15]: Isn't
```
[INFO] Including org.reactivestreams:reactive-streams:jar:1.0.2 in the shaded jar.
```
missing from this file?
(my mvn output is from 1.11 and a few days old 🤦 )

Comment (rmetzger) [reply flink-dist/src/main/resources/META-INF/NOTICE:15]: I have
```
[INFO] Including org.lz4:lz4-java:jar:1.5.0 in the shaded jar.
```
in my output, but this file lists version 1.6.0. Is this a 1.11 vs 1.12 issue?

File: flink-kubernetes/src/main/resources/META-INF/NOTICE [hunk 4]
```diff
@@ -11,15 +11,14 @@ This project bundles the following dependencies under the Apache Software Licens
...
-- com.squareup.okhttp3:logging-interceptor:3.12.0
+- com.squareup.okhttp3:logging-interceptor:3.12.6
 - com.squareup.okhttp3:okhttp:3.12.1
 - com.squareup.okio:okio:1.15.0
 - io.fabric8:kubernetes-client:4.9.2
 - io.fabric8:kubernetes-model:4.9.2
 - io.fabric8:kubernetes-model-common:4.9.2
 - io.fabric8:zjsonpatch:0.3.0
-- org.yaml:snakeyaml:1.23
+- org.yaml:snakeyaml:1.24
```
Comment (rmetzger) [inline flink-kubernetes/src/main/resources/META-INF/NOTICE:21]: ✅  Approved

File: flink-table/flink-table-planner-blink/src/main/resources/META-INF/NOTICE [hunk 5]
```diff
@@ -11,10 +11,6 @@ This project bundles the following dependencies under the Apache Software Licens
 - com.fasterxml.jackson.core:jackson-core:2.10.1
 - com.fasterxml.jackson.core:jackson-databind:2.10.1
 - com.jayway.jsonpath:json-path:2.4.0
-- joda-time:joda-time:2.5
-- net.minidev:json-smart:jar:2.3
-- net.minidev:accessors-smart:jar:1.2
-- org.ow2.asm:asm:jar:5.0.4
 - org.apache.calcite:calcite-core:1.22.0
 - org.apache.calcite:calcite-linq4j:1.22.0
 - org.apache.calcite.avatica:avatica-core:1.16.0
```
Comment (rmetzger) [inline flink-table/flink-table-planner-blink/src/main/resources/META-INF/NOTICE:16]: I also see 
```
[INFO] Including commons-io:commons-io:jar:2.4 in the shaded jar.
```

File: flink-dist/src/main/resources/META-INF/NOTICE (hunk 3 above)
Comment (zentol) [reply flink-dist/src/main/resources/META-INF/NOTICE:15]: lz4-java was bumped in https://github.com/apache/flink/commit/889705c7ab580737f0650f39dcd3712ec19c99e9 and is available in both master and release-1.11 . Was your branch possibly outdated?

Comment (zentol) [reply flink-dist/src/main/resources/META-INF/NOTICE:15]: reactive-streams  is at the very bottom ;)

File: flink-table/flink-table-planner-blink/src/main/resources/META-INF/NOTICE (hunk 5 above)
Comment (zentol) [reply flink-table/flink-table-planner-blink/src/main/resources/META-INF/NOTICE:16]: correct,  this is indeed missing

File: flink-dist/src/main/resources/META-INF/NOTICE (hunk 3 above)
Comment (rmetzger) [reply flink-dist/src/main/resources/META-INF/NOTICE:15]: Ah, I didn't know that you have to click this "expand" button multiple times. My bad.

Comment (rmetzger) [reply flink-dist/src/main/resources/META-INF/NOTICE:15]: > lz4-java was bumped in 889705c and is available in both master and release-1.11 . Was your branch possibly outdated?

Yes, I was outdated

File: flink-table/flink-table-planner-blink/src/main/resources/META-INF/NOTICE (hunk 5 above)
Comment (rmetzger) [reply flink-table/flink-table-planner-blink/src/main/resources/META-INF/NOTICE:16]: at least something ;)

**Pull request merged**